scripts/generate_pkcs7_tests.py

```diff
@@ -38,12 +38,7 @@ class TestData:
     Take in test_suite_pkcs7.data file.
     Allow for new tests to be added.
     """
-    #temporary solution to determine correct dependency macros between 3.6 and 4.0
-    #see issue #51 in mbedtls-framework
-    if build_tree.is_mbedtls_3_6():
-        mandatory_dep = "MBEDTLS_MD_CAN_SHA256"
-    else:
-        mandatory_dep = "PSA_WANT_ALG_SHA_256"
+    mandatory_dep = test_case.psa_or_3_6_feature_macro("PSA_ALG_SHA_256", "")
```

Comment:
As a matter of principle, we should indicate the domain as being `USE_PSA` (PKCS7, like all X.509, is in this domain), even it it turns out it doesn't matter for hashes.